sql/memo/rel_props.go

```diff
@@ -90,6 +91,7 @@ func newRelProps(rel RelExpr) *relProps {
 	default:
 	}
 
+	p.stat = statsForRel(rel)
```

Comment:
Does it matter that the statsForRel call happens before `populateFds` here, and after `populateFds` on line 58?

Should this be outlined?

Reply:
Pood point, we generate them lazily so as long as the children FDs are populated before this expr it ends up OK. I'll reorder the FDs to be first though.